### ISSUE TITLE
Error when building under macOS X with gcc

### ISSUE THREAD

**Issue by ilumary** [KhronosGroup/glslang#3062]
I am trying to build glslang under macOS x arm using gcc 12.2. It gives me the following error:
`aarch64-apple-darwin22-c++-12: error: unrecognized command-line option '-Xarch_arm64'`
Of course gcc uses the flag [-march=armv8-a](https://gcc.gnu.org/onlinedocs/gcc-11.2.0/gcc/ARM-Options.html#ARM-Options) but it still tries to use the clang flag.
My build command looks like this:
```
cmake -DCMAKE_INSTALL_PREFIX=extern/glslang/build \                      
      -DCMAKE_CXX_COMPILER=aarch64-apple-darwin22-c++-12 -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_INSTALL_PREFIX=extern/glslang/build/install \
      -Sextern/glslang \
      -Bextern/glslang/build
```
The build file inside the build directory (`build/glslang/CMakeFiles/MachineIndependent.dir/build.make`) is the only one generate with these flags. A simple search and replace with the correct flag fixes the issue.

**Comment by greg-lunarg** [comment]
Is this a Cmake bug? Does a newer version of CMake fix the problem?

**Comment by ilumary** [comment]
No it doesn't. I am using the lastest available stable version 3.24.3.

**Comment by juan-lunarg** [comment]
I'm unable to repro the build failure with my PR adding GCC to MacOS CI. 

If I had to guess your issue was only specifying the `CMAKE_CXX_COMPILER` and not the `CMAKE_C_COMPILER`. Hard to say without more reproducible information.
